generators/java/templates/src/main/java/_package_/_entityPackage_/domain/_persistClass_.java.jhi.ejs

```diff
@@ -343,7 +343,7 @@ _%>
             return false;
         }
 <%_ if (!embedded) { _%>
-        return <%= primaryKey.name %> != null && <%= primaryKey.name %>.equals(((<%= persistClass %>) o).<%= primaryKey.name %>);
+        return <%= primaryKey.name %> != null && <%= primaryKey.name %>.equals(((<%= persistClass %>) o).getId());
```

Comment:
The primary key might not be named `id`, so I think this needs to change to read from a variable.

Reply:
We can use getter like this:
```suggestion
        return <%= primaryKey.name %> != null && get<%= primaryKey.nameCapitalized %>.equals(((<%= persistClass %>) o).get<%= primaryKey.nameCapitalized %>());
```